pkg/admin/prerun/prerun.go

```diff
@@ -288,22 +281,67 @@ func (pr *PreRun) restore() error {
 	)
 	backupsForDeployment := getExistingBackupsForTheDeployment(existingBackups, currentDeploymentID)
 
-	if len(backupsForDeployment) == 0 {
-		return fmt.Errorf("there is no backup to restore for current deployment %q", currentDeploymentID)
-	}
-
 	if len(backupsForDeployment) > 1 {
 		// could happen during backing up when removing older backups failed
 		klog.InfoS("TODO: more than 1 backup, need to pick most recent one")
 	}
 
-	err = pr.dataManager.Restore(backupsForDeployment[0])
+	if len(backupsForDeployment) > 0 {
+		err = pr.dataManager.Restore(backupsForDeployment[0])
+		if err != nil {
+			return fmt.Errorf("failed to restore backup: %w", err)
+		}
+		klog.Info("Finished handling unhealthy system")
+		return nil
+	}
+
+	klog.InfoS("There is no backup to restore for current deployment - trying to restore backup for rollback deployment")
+	rollbackDeployID, err := getRollbackDeploymentID()
 	if err != nil {
-		return fmt.Errorf("failed to restore backup: %w", err)
+		return err
 	}
 
-	klog.Info("Finished restore")
-	return nil
+	if rollbackDeployID == "" {
+		// No backup for current deployment and there is no rollback deployment.
+		// This could be a unhealthy system that was manually rebooted to
+		// remediate the situation - let's not interfere: no backup, no restore, just proceed.
+		klog.InfoS("System has no rollback but health.json suggests system was rebooted - exiting prerun and proceeding with start")
```

Comment:
```suggestion
		klog.InfoS("System has no rollback but health.json suggests system was rebooted - skipping prerun")
```

This code should not assume what might come next, so let's just have the log messages comment on the local actions.